bin/node/runtime/src/impls.rs

```diff
@@ -281,6 +282,98 @@ impl liberland_traits::OnLLMPoliticsUnlock<AccountId32> for OnLLMPoliticsUnlock
 	}
 }
 
+#[derive(
+	Clone,
+	Copy,
+	Eq,
+	PartialEq,
+	Encode,
+	Decode,
+	RuntimeDebug,
+	MaxEncodedLen,
+	scale_info::TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Coords {
+	lat: u64,
+	long: u64,
+}
+
+#[derive(
+	Clone,
+	Encode,
+	Decode,
+	RuntimeDebug,
+	MaxEncodedLen,
+	scale_info::TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Metadata<MaxCoords: Get<u32>, MaxString: Get<u32>> {
+	demarcation: BoundedVec<Coords, MaxCoords>,
+	r#type: BoundedVec<u8, MaxString>,
+	status: BoundedVec<u8, MaxString>,
+}
+
+pub struct LandMetadataValidator;
+
+impl<StringLimit> pallet_nfts::traits::MetadataValidator<u32, u32, StringLimit>
+	for LandMetadataValidator
+{
+	fn validate_metadata(collection: u32, _: u32, data: &BoundedVec<u8, StringLimit>) -> bool {
+		// https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+		fn intersection(a: Coords, b: Coords, c: Coords, d: Coords) -> bool {
+			fn ccw(a: Coords, b: Coords, c: Coords) -> bool {
+				(b.long - a.long) * (c.lat - b.lat) > (b.lat - a.lat) * (c.long - b.long)
+			}
+			ccw(a,c,d) != ccw(b,c,d) && ccw(a,b,c) != ccw(a,b,d)
+		}
+
+		if collection != 1 { return true; }
+
+		let data = data.clone();
+		let data = Metadata::<ConstU32<100>, ConstU32<100>>::decode(&mut TrailingZeroInput::new(&data[..]));
+		if data.is_err() {
+			return false;
+		}
+
+		let data = data.unwrap();
+
+		// does it have at least 3 points
+		if data.demarcation.len() < 3 {
+			return false;
+		}
+
+		// is it roughly in the good place
+		for c in data.demarcation.iter() {
+			if c.lat  < 45_7686480 ||
+			   c.lat  > 45_7785989 ||
+			   c.long < 18_8821180 ||
+			   c.long > 18_8892809 {
+				return false;
+			}
+		}
+
+		let mut lines = vec![];
+		for i in 1..data.demarcation.len() {
+			lines.push((data.demarcation[i-1].clone(), data.demarcation[i].clone()))
+		}
+		lines.push((data.demarcation[data.demarcation.len()-1],data.demarcation[0]));
+
+		for i in 0..lines.len() {
+			let a = lines[i];
+			for j in i+2..lines.len()+i-1 {
+				let j = j%lines.len();
+				let b = lines[j];
+				if intersection(a.0, a.1, b.0, b.1) {
+					return false;
```

Comment:
this is self intersect ? Add comment explaining that its self intersect pls if it is